pandas/io/pytables.py

```diff
@@ -3363,7 +3366,11 @@ def read(
 
             columns = items[items.get_indexer(blk_items)]
             df = DataFrame(values.T, columns=columns, index=axes[1], copy=False)
-            if using_string_dtype() and is_string_array(values, skipna=True):
+            if (
+                using_string_dtype()
+                and isinstance(values, np.ndarray)
+                and is_string_array(values, skipna=True)
+            ):
```

Comment:
The same code pattern happens above in SeriesFixed `read()` method. Not entirely sure if the same change should be applied there as well, but would expect so (but maybe not covered by any test?)

Reply:
Indeed, added a test